sources/packages/backend/apps/api/src/route-controllers/application/_tests_/application.students.controller.getCompletedApplicationDetails.e2e-spec.ts

```diff
@@ -540,6 +560,338 @@ describe("ApplicationStudentsController(e2e)-getCompletedApplicationDetails", ()
+          msfaaNumber,
+        },
+        {
+          applicationStatus: ApplicationStatus.Completed,
+          offeringIntensity: OfferingIntensity.partTime,
+          firstDisbursementInitialValues: {
+            coeStatus: COEStatus.completed,
+            disbursementScheduleStatus: DisbursementScheduleStatus.Pending,
+          },
+        },
+      );
+      const [firstDisbursement] =
+        application.currentAssessment.disbursementSchedules;
+
+      application.currentAssessment.workflowData.calculatedData.pdppdStatus =
+        true;
+      await db.studentAssessment.save(application.currentAssessment);
+
+      application.student.disabilityStatus = DisabilityStatus.Requested;
+      await db.student.save(application.student);
+
+      const endpoint = `/students/application/${application.id}/completed`;
+      const token = await getStudentToken(
+        FakeStudentUsersTypes.FakeStudentUserType1,
+      );
+
+      // Act/Assert
+      await request(app.getHttpServer())
+        .get(endpoint)
+        .auth(token, BEARER_AUTH_TYPE)
+        .expect(HttpStatus.OK)
+        .expect({
+          firstDisbursement: {
+            coeStatus: firstDisbursement.coeStatus,
+            disbursementScheduleStatus:
+              firstDisbursement.disbursementScheduleStatus,
+          },
+          assessmentTriggerType: application.currentAssessment.triggerType,
+          hasBlockFundingFeedbackError: false,
+          eCertFailedValidations: [
+            ECertFailedValidation.DisabilityStatusNotConfirmed,
+          ],
+        });
+    },
+  );
+
+  it(
+    "Should get application details with ecert failed validations array having MSFAA not signed, cancelled and not valid when " +
+      "the the MSFAA number is not signed and there is a cancelled date and the offering intensity is part time.",
+    async () => {
+      // Arrange
+      const student = await saveFakeStudent(db.dataSource);
+      const msfaaNumber = createFakeMSFAANumber(
+        {
+          student,
+        },
+        {
+          msfaaState: MSFAAStates.Signed,
+        },
+      );
+      await db.msfaaNumber.save(msfaaNumber);
+
+      // Mock user services to return the saved student.
+      await mockUserLoginInfo(appModule, student);
+
+      const application = await saveFakeApplicationDisbursements(
+        appDataSource,
+        { student, msfaaNumber },
+        {
+          applicationStatus: ApplicationStatus.Completed,
+          offeringIntensity: OfferingIntensity.partTime,
+          firstDisbursementInitialValues: {
+            coeStatus: COEStatus.completed,
+            disbursementScheduleStatus: DisbursementScheduleStatus.Pending,
+          },
+        },
+      );
+      const [firstDisbursement] =
+        application.currentAssessment.disbursementSchedules;
+
+      msfaaNumber.dateSigned = null;
+      msfaaNumber.cancelledDate = getISODateOnlyString(new Date());
+      await db.msfaaNumber.save(msfaaNumber);
```

Comment:
The `createFakeMSFAANumber` allows the association of `MSFAAStates` as shown below.
```ts
const msfaaNumber = createFakeMSFAANumber(
  {
    student,
  },
  {
    msfaaState: MSFAAStates.Pending | MSFAAStates.CancelledOtherProvince,
  },
);
````